src/jedi_bundle/utils/git.py

```diff
@@ -66,29 +66,40 @@ def repo_is_reachable(logger, url, username, token):
 # --------------------------------------------------------------------------------------------------
 
 
-def repo_has_branch(logger, url, branch, is_tag=False):
+def repo_has_branch(logger, url, branch, is_tag=False, is_commit=False):
 
-    # Command to check if branch exists and pass exit code back
-    heads_or_tags = '--heads'
-    if is_tag:
-        heads_or_tags = '--tags'
+    if is_commit:
+        commit_url = url + '/commits/' + branch
+        r = requests.get(commit_url)
+        if r.ok:
+            logger.info(f'Found commit at {commit_url}')
+        else:
+            logger.info(f'Cannot find commit at {commit_url}')
+        return r.ok
 
-    git_ls_cmd = ['git', 'ls-remote', heads_or_tags, '--exit-code', url, branch]
+    else:
```

Comment:
I'm being a bit pedantic here, _but_ (in the interest of thorough code review): Because the earlier `if` statement _always_ ends with an early return, the `else` block here is unnecessary, so it can be removed and the code block below can be un-indented. (These are known as [guard](https://en.wikipedia.org/wiki/Guard_(computer_science)) statements and are a popular style recommendation for reducing indentation levels).